docs/docs/guides/web3_config/index.md

```diff
@@ -201,44 +202,394 @@ console.log(web3.getContextObject().config)
+The `transactionPollingTimeout` is used over HTTP connections. This option defines the number of seconds Web3 will wait for a receipt which confirms that a transaction was mined by the network. It can be set based on the average transaction confirmation time on the network. Note: If the `transactionPollingTimeout` is exceeded, the transaction may still be pending. 
+
+The default value of `transactionPollingTimeout` is 750 seconds (12.5 minutes).
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.transactionPollingTimeout = 750000; // 750000 ms = 750 s
+
+console.log(web3.getContextObject().config);
+```
+
+### [transactionReceiptPollingInterval](/api/web3-core/class/Web3Config#transactionReceiptPollingInterval)
+The `transactionReceiptPollingInterval` is used over HTTP connections. This option defines the number of seconds between Web3 calls for a receipt which confirms that a transaction was mined by the network. Compared to [`transactionPollingInterval`](/guides/web3_config/#transactionpollinginterval), it takes higher precedence. When this value is set, it will be read first.
+
+The default value of `transactionReceiptPollingInterval` is `undefined`.
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.transactionReceiptPollingInterval = undefined;
+
+console.log(web3.getContextObject().config);
+```
+
+### [transactionSendTimeout](/api/web3-core/class/Web3Config#transactionSendTimeout)
+The `transactionSendTimeout` option is used to specify how long to wait for the network to return the sent transaction result. Note: If the RPC call times out, the transaction may still be pending or even mined by the network. We recommend checking the pending transactions in such a case.
+
+The default value of `transactionSendTimeout` is 750 seconds (12.5 minutes).
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.transactionSendTimeout = 750000; // 750000 ms = 750 s
+
+console.log(web3.getContextObject().config);
+```
+
+### [transactionConfirmationPollingInterval](/api/web3-core/class/Web3Config#transactionConfirmationPollingInterval)
+The `transactionConfirmationPollingInterval` option is deprecated. Please use [`transactionReceiptPollingInterval`](/guides/web3_config/#transactionreceiptpollinginterval) or [`transactionPollingInterval`](/guides/web3_config/#transactionpollinginterval) instead.
+
+### [blockHeaderTimeout](/api/web3-core/class/Web3Config#blockHeaderTimeout)
+The `blockHeaderTimeout` is used over socket-based connections. After sending a transaction, it will listen for the appearance of new blocks and proceed with subsequent operations based on the transaction results within them. This option defines the amount seconds it should wait for 'newBlockHeaders' event before falling back to polling to fetch transaction receipt.
```

Comment:
in case if subscription fails it auto reverts to polling for looking blockHeaderTimeout